epochStart/bootstrap/process.go

```diff
@@ -894,10 +884,58 @@ func (e *epochStartBootstrap) syncUserAccountsState(rootHash []byte) error {
 	return nil
 }
 
+func (e *epochStartBootstrap) createStorageService(
+	shardCoordinator sharding.Coordinator,
+	pathManager storage.PathManagerHandler,
+	epochStartNotifier storage.EpochStartNotifier,
+	startEpoch uint32,
+	createTrieEpochRootHashStorer bool,
+	targetShardId uint32,
+) (dataRetriever.StorageService, error) {
+	storageServiceCreator, err := storageFactory.NewStorageServiceFactory(
+		&e.generalConfig,
+		&e.prefsConfig,
+		shardCoordinator,
+		pathManager,
+		epochStartNotifier,
+		e.coreComponentsHolder.NodeTypeProvider(),
+		startEpoch,
+		createTrieEpochRootHashStorer,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if targetShardId == core.MetachainShardId {
+		return storageServiceCreator.CreateForMeta()
+	}
+
+	return storageServiceCreator.CreateForShard()
+}
+
 func (e *epochStartBootstrap) createTriesComponentsForShardId(shardId uint32) error {
 	e.tryCloseExisting(factory.UserAccountTrie)
 	e.tryCloseExisting(factory.PeerAccountTrie)
 
+	storageService, err := e.createStorageService(
+		e.genesisShardCoordinator,
+		e.coreComponentsHolder.PathHandler(),
+		&disabled.EpochStartNotifier{},
+		e.startEpoch,
+		false,
+		shardId,
+	)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
```

Comment:
is it ok to call defer here? It will close the underlying storers that the trie pointers use.